lib/paper_trail/attribute_serializers/cast_attribute_serializer.rb

```diff
@@ -32,6 +32,10 @@ def deserialize(attr, val)
         if defined_enums[attr] && val.is_a?(::String)
           # Because PT 4 used to save the string version of enums to `object_changes`
           val
+        elsif PaperTrail::RAILS_GTE_7_0 && val.is_a?(ActiveRecord::Type::Time::Value)
+          # Because Rails 7 time attribute throws a delegation error when you deserialize
+          # it with the factory.
```

Comment:
Should we add a link to https://github.com/rails/rails/issues/43966 to this comment?

Reply:
Done!